force-app/main/default/classes/Grid.cls

```diff
@@ -0,0 +1,124 @@
+/**
+* @description tracks the position of a user in a 5x5 grid as x and y coordinates. 
+* The coordinates are index origin 0. The default position is the top left,
+* but the user can specify the position in the constructor.
+*/
+public with sharing class Grid 
+{
+    // holds the x coordinate within the 5x5 grid
+    public Integer x
+    {
+        get { return x; }
+        set 
+        { 
+            if (value < 0) 
+            {
+                x = 0;
+            } 
+            else if (value > 4)
+            {
+                x = 4;
+            }
+            else
+            {
+                x = value;
+            }
+        }
+    }
+
+    // holds the y coordinate within the 5x5 grid
+    public Integer y
+    {
+        get { return y; }
+        set 
+        { 
+            if (value < 0) 
+            {
+                y = 0;
+            } 
+            else if (value > 4)
+            {
+                y = 4;
+            }
+            else
+            {
+                y = value;
+            }
+        }
+    }
+
+    // no-arguent constructor, x,y = (0,0) 
+    public Grid() 
+    {
+        This.x = 0;
+        This.y = 0;
+    }
+    
+    // constructor, specify initial x,y coordinates
+    public Grid(Integer x, Integer y)
+    {
+        This.x = x;
+        This.y = y;
+    }
+
+    // move position (x-1,y), if within 5x5 grid
+    public void left()
+    {
+        if (x > 0)
+        {
+            x--;
```

Comment:
Wouldn't this be handled by the setter? i guess it might save a cpu cycle or two when x is 0, but it means you are checking in multiple places so that would consume some.

Reply:
ah, yes, I see, i forgot that the setter would be called as its updating the variable (sounds obvious now I say it out loud!)